src/policy_artifacthub.rs

```diff
@@ -0,0 +1,923 @@
+                    String::from("https://github.com/home"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_POLICY_OCIURL),
+                    String::from("https://github.com/ocirepo"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_POLICY_SOURCE),
+                    String::from("https://github.com/repo"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_POLICY_LICENSE),
+                    String::from("Apache-2.0"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_POLICY_USAGE),
+                    String::from("readme contents"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_ARTIFACTHUB_RESOURCES),
+                    String::from("Pod, Deployment"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_ARTIFACTHUB_DISPLAYNAME),
+                    String::from("Verify Image Signatures"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_ARTIFACTHUB_KEYWORDS),
+                    String::from("pod, signature"),
+                ),
+                (
+                    String::from(KUBEWARDEN_ANNOTATION_RANCHER_HIDDENUI),
+                    String::from("true"),
+                ),
+            ])),
+            mutating: false,
+            background_audit: true,
+            context_aware: false,
+            execution_mode: Default::default(),
+        }
+    }
+
+    #[test]
+    fn artifacthubpkg_validate_inputs() -> Result<(), ()> {
+        // check annotations None
+        let arthub = ArtifactHubPkg::from_metadata(
+            &Metadata::default(),
+            "0.2.1",
+            OffsetDateTime::UNIX_EPOCH,
+            None,
+        );
+        assert_eq!(
+            arthub.unwrap_err().to_string(),
+            "no annotations in policy metadata. policy metadata must specify annotations"
+        );
+
+        // check annotations empty
+        let metadata = Metadata {
+            annotations: Some(HashMap::from([])),
+            ..Default::default()
+        };
+        let arthub =
+            ArtifactHubPkg::from_metadata(&metadata, "0.2.1", OffsetDateTime::UNIX_EPOCH, None);
+        assert_eq!(
+            arthub.unwrap_err().to_string(),
+            "no annotations in policy metadata. policy metadata must specify annotations"
+        );
+
+        // check version is semver
+        let arthub = ArtifactHubPkg::from_metadata(
+            &mock_metadata_with_minimum_required(),
+            "not-semver",
+            OffsetDateTime::UNIX_EPOCH,
+            None,
+        );
+        assert_eq!(
+            arthub.unwrap_err().to_string(),
+            "policy version must be in semver: unexpected character 'n' while parsing major version number"
+        );
```

Comment:
It's not your fault, there's no other way to do that... But I think we should really get rid of `anyhow` and use `thiserror`.

I see two possibilities:

* We leave the code as it is, we merge this PR and later we do a big PR that kills `anyhow::Result`
* You change this PR: you introduce a first version of our `Error` type, with just the errors that the code of this PR introduces. You refactor the tests to make use of this Error. Later we do another PR that extends the custom error you introduced and just removes anyhow

Reply:
Let's introduce `thiserror` in a single shot, later on

Reply:
Marking as unresolved, now that you have introduced `thiserror` the unit tests should look for the error type, not for its string representation